hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/HoodieParquetFileFormat.scala

```diff
@@ -34,6 +34,15 @@ class HoodieParquetFileFormat extends ParquetFileFormat with SparkAdapterSupport
 
   override def toString: String = "Hoodie-Parquet"
 
+  override def supportBatch(sparkSession: SparkSession, schema: StructType): Boolean = {
+    if (HoodieSparkUtils.gteqSpark3_4) {
```

Comment:
Im wondering if this should be for greater than equal to spark 332, so that we can have this fix there as well.

Reply:
Let's take this separately since this PR is not related to Spark 3.3.2.  The code here is to make sure all other versions maintain the same logic.

Reply:
The tests fail for other spark versions if I don't add this check.
```
Merge Hudi to Hudi *** FAILED ***
2023-06-06T23:38:24.7660935Z   org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 3194.0 failed 1 times, most recent failure: Lost task 0.0 in stage 3194.0 (TID 3768) (fv-az1128-658 executor driver): java.lang.ClassCastException: org.apache.spark.sql.vectorized.ColumnarBatchRow cannot be cast to org.apache.spark.sql.vectorized.ColumnarBatch
2023-06-06T23:38:24.7662056Z 	at org.apache.spark.sql.execution.FileSourceScanExec$$anon$1.next(DataSourceScanExec.scala:560)
2023-06-06T23:38:24.7662628Z 	at org.apache.spark.sql.execution.FileSourceScanExec$$anon$1.next(DataSourceScanExec.scala:549)
2023-06-06T23:38:24.7663391Z 	at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:486)
```